ffi/dune

```diff
@@ -0,0 +1,20 @@
+(library
+ (name yaml_ffi)
+ (public_name yaml.ffi)
+ (libraries sexplib)
+ (foreign_archives ../vendor/yaml_c)
+ (ctypes
+  (external_library_name yaml)
+  (build_flags_resolver
+   (vendored
+     ; hack: repeated include dirs from root and relative since this gets
+     ;  called from multiple places
+    (c_flags (-I../vendor/headers -Ivendor/headers))))
```

Comment:
I think the only issue is here: this doesn't work in a dune workspace created with opam-monorepo. 
This is already known but I'm linking the relevant issue : https://github.com/ocaml/dune/issues/5325